src/app/doctor/services/usecases_test.go

```diff
@@ -0,0 +1,209 @@
+		mockRepo.On("SelectAllData").Return(sampleDomainList, nil).Once()
+		result, err := services.GetAllDoctors()
+
+		assert.Nil(t, err)
+		assert.Greater(t, len(result), 0)
+	})
+
+	t.Run("should got server error", func(t *testing.T) {
+		mockRepo.On("SelectAllData").Return(nil, errors.New("can't connect to the database")).Once()
+		result, err := services.GetAllDoctors()
+
+		assert.NotNil(t, err)
+		assert.Nil(t, result)
+	})
+}
+
+func TestGetDoctorByID(t *testing.T) {
+	t.Run("should got data by id", func(t *testing.T) {
+		mockRepo.On("SelectDataByID", sampleUUIDIkoUwais.String()).Return(&sampleDomainIkoUwais, nil).Once()
+		result, err := services.GetDoctorByID(sampleUUIDIkoUwais.String())
+
+		assert.Nil(t, err)
+		assert.Equal(t, sampleDomainIkoUwais.Name, result.Name)
+	})
+
+	t.Run("should got database error", func(t *testing.T) {
+		mockRepo.On("SelectDataByID", sampleUUIDIkoUwais.String()).Return(nil, errors.New("can't connect to the database")).Once()
+		result, err := services.GetDoctorByID(sampleUUIDIkoUwais.String())
+
+		assert.NotNil(t, err)
+		assert.Nil(t, result)
+	})
+
+	t.Run("should got error while data not found", func(t *testing.T) {
+		mockRepo.On("SelectDataByID", sampleUUIDIkoUwais.String()).Return(nil, errors.New("record not found"))
+		result, err := services.GetDoctorByID(sampleUUIDIkoUwais.String())
+
+		assert.NotNil(t, err)
+		assert.Nil(t, result)
+	})
+}
+
+func TestCreateDoctor(t *testing.T) {
+	t.Run("should created a data", func(t *testing.T) {
+		mockRepo.On("InsertData", sampleDomainIkoUwaisWithNoUUID).Return(sampleUUIDIkoUwais.String(), nil).Once()
+		result, err := services.CreateDoctor(sampleDomainIkoUwaisWithNoUUID)
+
+		assert.Nil(t, err)
+		assert.Equal(t, sampleUUIDIkoUwais.String(), result)
+	})
+
+	t.Run("should got database error", func(t *testing.T) {
+		mockRepo.On("InsertData", sampleDomainIkoUwaisWithNoUUID).Return(uuid.Nil.String(), errors.New("can't connect to the database")).Once()
+		result, err := services.CreateDoctor(sampleDomainIkoUwaisWithNoUUID)
+
+		assert.NotNil(t, err)
+		assert.Equal(t, uuid.Nil.String(), result)
+	})
+}
+
+func TestUpdateDoctorByID(t *testing.T) {
+	t.Run("should update data by id", func(t *testing.T) {
+		mockRepo.On("UpdateByID", sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID).Return(nil).Once()
+		err := services.AmendDoctorByID(sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID)
+
+		assert.Nil(t, err)
+	})
+
+	t.Run("should got database error", func(t *testing.T) {
+		mockRepo.On("UpdateByID", sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID).Return(errors.New("can't connect to the database")).Once()
+		err := services.AmendDoctorByID(sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID)
+
+		assert.NotNil(t, err)
+	})
+
+	t.Run("should got not found error", func(t *testing.T) {
+		mockRepo.On("UpdateByID", sampleUUIDIkoUwais.String(), sampleDomainIkoUwaisWithNoUUID).Return(errors.New("record not found")).Once()

```

Comment:
prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code